RecoEgamma/EgammaPhotonProducers/src/ConversionTrackProducer.cc

```diff
@@ -101,11 +101,6 @@ void ConversionTrackProducer::produce(edm::Event& e, const edm::EventSetup& es)
   edm::ESHandle<MagneticField> magFieldHandle;
   es.get<IdealMagneticFieldRecord>().get(magFieldHandle);
 
-  if (filterOnConvTrackHyp && !beamSpotHandle.isValid()) {
-    edm::LogError("Invalid Collection") << "invalid collection for the BeamSpot";
-    throw;
-  }
```

Comment:
The `beamspotHandle->position()` on line 114/109 will throw an exception (with much clearer message) if the Handle is not valid.

(by the way, does
```cpp
math::XYZVector beamSpot = math::XYZVector(beamSpotHandle->position());
```
really need to be repeated for each input track?)